# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/AbstractRange.h
# Commit: 4f30d7afdf22
# Full Hash: 4f30d7afdf22c652481513564b60b37053c0948b
# Author: Atila Butkovits <abutkovits@mozilla.com>
# Date: 2025-05-08 04:32:48
# Regressor Bug: 1932150
# File Overlap Count: 2
# Description:
#   Revert "Bug 1932150 - Fix build bustage" for causing build bustages.
#   
#   This reverts commit 0681b181d6a8a48ac460a97dacb7fca91f9989c3.
#   
#   Revert "Bug 1932150 - Fix lint failure" for causing Build bustage.
# ==============================================================================

diff -r 1e64a48e3517 -r 4f30d7afdf22 dom/base/AbstractRange.h
--- a/dom/base/AbstractRange.h	Wed May 07 20:11:46 2025 +0300
+++ b/dom/base/AbstractRange.h	Wed May 07 20:37:37 2025 +0300
@@ -30,7 +30,6 @@
 class Document;
 class Selection;
 class StaticRange;
-class HTMLSlotElement;
 
 enum class AllowRangeCrossShadowBoundary : bool { No, Yes };
 
@@ -194,9 +193,7 @@
             typename RangeType>
   static nsresult SetStartAndEndInternal(
       const RangeBoundaryBase<SPT, SRT>& aStartBoundary,
-      const RangeBoundaryBase<EPT, ERT>& aEndBoundary, RangeType* aRange,
-      AllowRangeCrossShadowBoundary aAllowCrossShadowBoundary =
-          AllowRangeCrossShadowBoundary::No);
+      const RangeBoundaryBase<EPT, ERT>& aEndBoundary, RangeType* aRange);
 
   template <class RangeType>
   static bool MaybeCacheToReuse(RangeType& aInstance);
@@ -230,15 +227,8 @@
 
   void UpdateCommonAncestorIfNecessary();
 
-  static void MarkDescendants(nsINode& aNode);
-  static void UnmarkDescendants(nsINode& aNode);
-
-  static void UpdateDescendantsInFlattenedTree(nsINode& aNode,
-                                               bool aMarkDescendants);
-  friend void mozilla::SlotAssignedNodeAdded(dom::HTMLSlotElement* aSlot,
-                                             nsIContent& aAssignedNode);
-  friend void mozilla::SlotAssignedNodeRemoved(dom::HTMLSlotElement* aSlot,
-                                               nsIContent& aUnassignedNode);
+  static void MarkDescendants(const nsINode& aNode);
+  static void UnmarkDescendants(const nsINode& aNode);
 
  private:
   void ClearForReuse();